src/main/java/net/findeasily/website/controller/mgmt/ListingController.java

```diff
@@ -105,7 +106,11 @@ public String createListingHandler(CurrentUser user,
             log.error("errors happen when creating new listing...");
         }
         Listing listing = listingService.save(form, user.getId());
-        return "redirect:/mgmt/listing/" + listing.getId();
+        // TODO: show error message to user
+        if (listing == null || StringUtils.isBlank(listing.getId())) {
```

Comment:
Hello, I got question, So here listing == null means that creating listing fails and what about : StringUtils.isBlank(listing.getId()), means the created listing is wrong?

Reply:
This is for defensive coding. Because in the following line `listing.getId()` will be called, if it is null, then NullPointException(NPE) will be thrown. To avoid this NPE, I did that check together with `list == null`.